bridge-fungible-token/src/utils.sw

```diff
@@ -92,18 +122,23 @@ pub fn adjust_deposit_decimals(msg_val: b256) -> Result<u64, BridgeFungibleToken
         if decimal_diff > 19u8 {
             return Result::Err(BridgeFungibleTokenError::BridgedValueIncompatability);
         };
-        let adjustment_factor = 10.pow(LAYER_1_DECIMALS - DECIMALS);
-        let bn_factor = U256::from((0, 0, 0, adjustment_factor));
-        let adjusted = value.divide(bn_factor);
 
-        let result = shift_decimals_left(adjusted, decimal_diff);
+        let adjustment_factor = U256::from((0, 0, 0, 10u32.pow(LAYER_1_DECIMALS - DECIMALS)));
+        let result = shift_decimals_right(value, decimal_diff);
+
         if result.is_err() {
             return Result::Err(BridgeFungibleTokenError::BridgedValueIncompatability);
         };
 
-        if result.unwrap() == value
-            && (value > bn_factor
-            || value == bn_factor)
+        let (adjusted, remainder) = result.unwrap();
+        // ensure that the value does not use higher precision than is bridgeable by this contract
+        if remainder != 0u32 {
```

Comment:
As noted, this check should probably be part of the logic in `shift_decimals_right` so it doesn't have to even return the remainder.

Reply:
I was thinking that by checking the remainder here and ensuring that it's `0`, I can avoid the need to perform the multiplication step I was previously doing. 
that is:
- what I want to achieve is checking that `A mod B == 0`.
- Previously the only way to do this was to check that  `(A / B) * A == 0`.
- Now, by returning the remainder I can do `let (q, r) = A / B`. If `r == 0`, I can proceed with using `q`.